src/hotspot/share/opto/loopnode.cpp

```diff
@@ -5011,6 +5011,22 @@ Node *PhaseIdealLoop::get_late_ctrl( Node *n, Node *early ) {
           }
         }
       }
+      // For Phis only consider Region's inputs that were reached by following the memory edges
+      if (LCA != early) {
+        for (uint i = 0; i < worklist.size(); i++) {
+          Node* s = worklist.at(i);
+          if (s->is_Phi() && C->can_alias(s->adr_type(), load_alias_idx)) {
+            Node* r = s->in(0);
+            for (uint j = 1; j < s->req(); j++) {
+              Node* in = s->in(j);
+              Node* r_in = r->in(j);
+              if (worklist.member(in) && is_dominator(early, r_in)) {
```

Comment:
Above we check for `!sctrl->is_top()`, do we need the same check for `r_in`?

Reply:
I don't think a Region can have a top input at this point (previous IGVN should have taken care of it).